src/pages/question/PersonaQuestion.tsx

```diff
@@ -1,9 +1,44 @@
-import React, { ReactElement } from 'react';
+import React, { ReactElement, useEffect, useState } from 'react';
+import { useHistory } from 'react-router-dom';
 
+import Card from './card';
+import questions from './questions';
 import * as S from './styled';
 
-type PersonaQuestionPageProps = {};
+export default function PersonaQuestionPage(): ReactElement {
+  const history = useHistory();
 
-export default function PersonaQuestionPage(_: PersonaQuestionPageProps): ReactElement {
-  return <S.Container>PersonaQuestionPage</S.Container>;
+  const [questionNum, setQuestionNum] = useState(1);
+  const [quesitonLen] = useState(questions.length);
+  const [question, setQuestion] = useState(questions[0]);
+
+  useEffect(() => {
+    questionNum > quesitonLen ? history.push('/persona/result') : '';
+    setQuestion(questions[questionNum - 1]);
+  }, [questionNum]);
+
+  const handleCardClick = () => {
+    setQuestionNum(questionNum + 1);
+  };
```

Comment:
useEffect 내부가 아니라 여기서 분기처리를 하는 것이 좋지 않을까요?

```suggestion
  const handleCardClick = () => {
    if (questionIdx < questions.length -1) {
    setQuestionNum(questionNum + 1); 
    } else {
    // 저장하고 다음 페이지로 이동
    }
  };
```